### ISSUE TITLE
Change the locale in which ProFTPD produces dates in log files

### ISSUE THREAD

**Issue by C-Duv** [proftpd/proftpd#544]
**TL;DR:** ProFTPD produces logs with dates in system's locale (eg. French: *juil. 28 16:42:17*) but `fail2ban` only recognizes dates in english locale (eg. *Jul 28 16:42:17*)
**How to make ProFTPD produce dates in "english" locale as Syslog, Apache and others do?**

### What I Did

Installed ProFTPD  v1.3.4a (via `proftpd-basic` Debian package) and enabled `fail2ban` *proftpd* jail.

Tested multiple dummy access with:

```Shell
for i in {1..20} ; do
    echo -n "$i: "
    curl ftp://foo:bar@server-hostname/baz-$i -s -S
done
```

### What I Expected/Wanted

Get blocked by `fail2ban` after about 10 attempts.

### What Happened

I did not get blocked, all my 20 requests went through: client-side I got  the `curl: (67) Access denied: 530` 20 times.

File `/var/log/proftpd/proftpd.log` got filled with:
```
juil. 28 16:42:17 server-hostname proftpd[9423] server-hostname (client-hostname[2.2.2.2]): FTP session opened.
juil. 28 16:42:17 server-hostname proftpd[9423] server-hostname (client-hostname[2.2.2.2]): USER foo: no such user found from client-hostname [2.2.2.2] to ::ffff:1.1.1.1:21
juil. 28 16:42:17 server-hostname proftpd[9423] server-hostname (client-hostname[2.2.2.2]): FTP session closed.
juil. 28 16:42:17 server-hostname proftpd[9424] server-hostname (client-hostname[2.2.2.2]): FTP session opened.
juil. 28 16:42:17 server-hostname proftpd[9424] server-hostname (client-hostname[2.2.2.2]): USER foo: no such user found from client-hostname [2.2.2.2] to ::ffff:1.1.1.1:21
juil. 28 16:42:17 server-hostname proftpd[9424] server-hostname (client-hostname[2.2.2.2]): FTP session closed.
```

### What Happened

`fail2ban` failed to recognized/parse log files because of the date is in French (*juil.* is for *juillet* witch is French for *july*).
```
fail2ban-regex /var/log/proftpd/proftpd.log /etc/fail2ban/filter.d/proftpd.conf
fail2ban-regex \
    "$(grep 'no such user found' /var/log/proftpd/proftpd.log)" \
    /etc/fail2ban/filter.d/proftpd.conf
```
Returns no match.

But this one does:
```
fail2ban-regex \
    "$( \
        grep 'no such user found' /var/log/proftpd/proftpd.log \
        | sed 's/^juil\./Jul/' \
    )" \
    /etc/fail2ban/filter.d/proftpd.conf
```

### ProFTPD Version and Configuration

Please help us reproduce the problem/issue you are encountering.  To do this,
we need to know which version of ProFTPD you are using, how it was built,
_etc_.  The following command is an easy way to get all of this information:

`proftpd -V`:
```
Compile-time Settings:
  Version: 1.3.4a (maint)
  Platform: LINUX [Linux 3.2.0-4-amd64 x86_64]
  Built: Tue May 19 2015 11:02:49 UTC
  Built With:
    configure  'CFLAGS=-g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security' 'CPPFLAGS=-D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security' 'FFLAGS=-g -O2' 'LDFLAGS=-Wl,-z,relro' '--prefix=/usr' '--with-includes=/usr/include/postgresql:/usr/include/mysql' '--mandir=/usr/share/man' '--sysconfdir=/etc/proftpd' '--localstatedir=/var/run' '--libexecdir=/usr/lib/proftpd' '--enable-sendfile' '--enable-facl' '--enable-dso' '--enable-autoshadow' '--enable-ctrls' '--with-modules=mod_readme' '--enable-ipv6' '--enable-nls' '--with-lastlog=/var/log/lastlog' '--enable-pcre' '--build' 'x86_64-linux-gnu' '--with-shared=mod_unique_id:mod_site_misc:mod_load:mod_ban:mod_quotatab:mod_sql:mod_sql_mysql:mod_sql_postgres:mod_sql_sqlite:mod_sql_odbc:mod_dynmasq:mod_quotatab_sql:mod_ldap:mod_quotatab_ldap:mod_ratio:mod_tls:mod_rewrite:mod_radius:mod_wrap:mod_wrap2:mod_wrap2_file:mod_wrap2_sql:mod_quotatab_file:mod_quotatab_radius:mod_facl:mod_ctrls_admin:mod_copy:mod_deflate:mod_ifversion:mod_tls_memcache:mod_exec:mod_sftp:mod_sftp_pam:mod_sftp_sql:mod_shaper:mod_sql_passwd:mod_ifsession' 'build_alias=x86_64-linux-gnu'

  CFLAGS: -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wall
  LDFLAGS: -L$(top_srcdir)/lib -Wl,-z,relro  -L/usr/lib/x86_64-linux-gnu -L/usr/lib
  LIBS: -lacl  -lpcreposix -lpcre -lssl -lcrypto -lcap  -lpam -lsupp -lcrypt -ldl  -ldl -lz

  Files:
    Configuration File:
      /etc/proftpd/proftpd.conf
    Pid File:
      /var/run/proftpd.pid
    Scoreboard File:
      /var/run/proftpd.scoreboard
    Header Directory:
      /usr/include/proftpd
    Shared Module Directory:
      /usr/lib/proftpd

  Features:
    + Autoshadow support
    + Controls support
    + curses support
    - Developer support
    + DSO support
    + IPv6 support
    + Largefile support
    + Lastlog support
    - Memcache support
    + ncurses support
    + NLS support
    + OpenSSL support
    + PCRE support
    + POSIX ACL support
    + Shadow file support
    + Sendfile support
    + Trace support

  Tunable Options:
    PR_TUNABLE_BUFFER_SIZE = 1024
    PR_TUNABLE_DEFAULT_RCVBUFSZ = 8192
    PR_TUNABLE_DEFAULT_SNDBUFSZ = 8192
    PR_TUNABLE_GLOBBING_MAX_MATCHES = 100000
    PR_TUNABLE_GLOBBING_MAX_RECURSION = 8
    PR_TUNABLE_HASH_TABLE_SIZE = 40
    PR_TUNABLE_NEW_POOL_SIZE = 512
    PR_TUNABLE_SCOREBOARD_BUFFER_SIZE = 80
    PR_TUNABLE_SCOREBOARD_SCRUB_TIMER = 30
    PR_TUNABLE_SELECT_TIMEOUT = 30
    PR_TUNABLE_TIMEOUTIDENT = 10
    PR_TUNABLE_TIMEOUTIDLE = 600
    PR_TUNABLE_TIMEOUTLINGER = 30
    PR_TUNABLE_TIMEOUTLOGIN = 300
    PR_TUNABLE_TIMEOUTNOXFER = 300
    PR_TUNABLE_TIMEOUTSTALLED = 3600
    PR_TUNABLE_XFER_SCOREBOARD_UPDATES = 10
```

## System configuration

`locale`:
```
LANG=fr_FR.UTF-8
LANGUAGE=
LC_CTYPE="fr_FR.UTF-8"
LC_NUMERIC="fr_FR.UTF-8"
LC_TIME="fr_FR.UTF-8"
LC_COLLATE="fr_FR.UTF-8"
LC_MONETARY="fr_FR.UTF-8"
LC_MESSAGES="fr_FR.UTF-8"
LC_PAPER="fr_FR.UTF-8"
LC_NAME="fr_FR.UTF-8"
LC_ADDRESS="fr_FR.UTF-8"
LC_TELEPHONE="fr_FR.UTF-8"
LC_MEASUREMENT="fr_FR.UTF-8"
LC_IDENTIFICATION="fr_FR.UTF-8"
LC_ALL=
```

**Comment by Castaglia** [comment]
On startup, ProFTPD explicitly sets the `LC_ALL` environment variable when the `--enable-nls` compile option is used; this is so that the `LC_MESSAGES` and `LC_COLLATE` (and `LC_MONETARY`) environment variables can be set _for sessions_ by the `mod_lang` module.

You might try adding the following to your `proftpd.conf`, outside of any `<Global>` and `<VirtualHost>` sections:

    # The LC_TIME environment variable is used for formatting dates via e.g. strftime(3)
    SetEnv LC_TIME fr_FR.UTF-8

Another thing to note is that the logging format _changed_ in later versions of ProFTPD, to instead be:

    2017-07-28 16:42:17 server-hostname proftpd[9423] server-hostname (client-hostname[2.2.2.2]):

As the root cause is actually `fail2ban`'s _inability_ to handle localized text, I'm not feeling like changing the ProFTPD code actually helps; it doesn't address the root problem.

**Comment by C-Duv** [comment]
I added `SetEnv LC_TIME en_US.UTF-8` (under the `Port` directive) and it did not made the date in log files go english.

I've found (with help from #debian IRC channel) the following to work (on Debian obviously):
```Shell
cat >> /etc/default/proftpd <<'EOT'

# Force dates' language to en_US.UTF-8 (to be fail2ban-friendly):
export LC_TIME=en_US.UTF-8
EOT
```

I understand that `fail2ban` is in cause, I was a bit surprised to see french dates in my logs ;)
Newer format is great!

**Comment by Castaglia** [comment]
Great, I'm glad you were able to get it working!  I've added the above as a note in the [Logging howto](https://github.com/proftpd/proftpd/commit/7df27542effdcdba9ff3caf1967bcd818f6cf15e), for any future readers.

**Comment by C-Duv** [comment]
Great :+1:
